cmd/restic/global.go

```diff
@@ -111,8 +121,14 @@ func init() {
 	f.BoolVar(&globalOptions.CleanupCache, "cleanup-cache", false, "auto remove old cache directories")
 	f.IntVar(&globalOptions.LimitUploadKb, "limit-upload", 0, "limits uploads to a maximum rate in KiB/s. (default: unlimited)")
 	f.IntVar(&globalOptions.LimitDownloadKb, "limit-download", 0, "limits downloads to a maximum rate in KiB/s. (default: unlimited)")
+	f.UintVar(&globalOptions.MinPackSize, "min-packsize", 0, "set min pack size in MiB. (default: $RESTIC_MIN_PACKSIZE or 4)")
 	f.StringSliceVarP(&globalOptions.Options, "option", "o", []string{}, "set extended option (`key=value`, can be specified multiple times)")
 
+	//setting default to 0 for these then plugging default values later fixes text output for (default: )
+	if globalOptions.MinPackSize == 0 {
+		globalOptions.MinPackSize = uint(minpacksize)
+	}
```

Comment:
> We need some upper limit for the pack size, especially we don't want to cross the gigabyte boundary.

Fully agree! In fact the **maximum** pack size should be small enough such that it can be stored in a `uint32`... (to save some memory when saving `offset`)

Reply:
Our hard limit if we want uint32 would be 4095MB.
We don't want to cross the GB boundary, so can I get a sanity check on a 1023MB limit?

I don't know if we want to mess with converting to uint32 right now, because that might effect the repository format in a manner could cause problems.

Reply:
I'm not sure whether that limit is strict enough. I wonder whether this causes memory usage problems for commands like `check` or `prune`.

Reply:
What would you consider to be a memory usage problem?  I think properly documented memory usage estimates (or an inline warning, like I've implemented on save blob and file read concurrency in cmd_backup.go line 163) would be preferable to a hard limit.  Some people will consider 1GB of memory usage a problem, others might consider 64GB to be perfectly acceptable.

I'm not sure what the project's preferences are when it comes to configurability.  I'm of the opinion that we should allow the user to do whatever they want, as long as they're made aware of potential drawbacks.  

I'm happy to run some tests on crazy large pack sizes for a prune and check, once I get my NAS fixed.